src/Agent.Worker/Build/TfsVCTeeSourceProvider.cs

```diff
@@ -121,12 +122,66 @@ public sealed class TfsVCTeeSourceProvider : SourceProvider, ISourceProvider
                 }
             }
 
-            // Sync the sources.
-            // TODO: build.sourceTfvcShelveset
-            // TODO: Gated.
+            // Get.
```

Comment:
I need to follow-up with Patrick about the sequence here. After a bit of testing, I ended up going with Get and then Unshelve. The reverse order omits support for a specific scenario: build a shelveset against a specific commit.

This order appears to be safe. TEE's TF returns non-zero if something can't cleanly autoresolve. I need to find out if there are any scenarios where this isn't true.